bee-network/src/swarm/protocols/gossip/gossip_upgrade.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::init::global::network_id;
+
+use futures::{future, AsyncRead, AsyncWrite};
+use libp2p::{core::UpgradeInfo, InboundUpgrade, OutboundUpgrade};
+use log::trace;
+
+use std::iter;
+
+/// Configuration for an upgrade to the `IotaGossip` protocol.
+#[derive(Debug, Clone, Default)]
+pub struct GossipUpgrade;
+
+impl UpgradeInfo for GossipUpgrade {
+    type Info = Vec<u8>;
+    type InfoIter = iter::Once<Self::Info>;
+
+    fn protocol_info(&self) -> Self::InfoIter {
+        iter::once(format!("/iota-gossip/{}/1.0.0", network_id()).as_bytes().to_vec())
```

Comment:
`String` has an `into_bytes` method: https://doc.rust-lang.org/std/string/struct.String.html#method.into_bytes

Use this to avoid a second allocation and reduce complexity.